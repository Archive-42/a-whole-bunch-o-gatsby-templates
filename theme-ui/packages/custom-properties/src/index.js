import pluralize from 'pluralize'

export default (theme, prefix) => {
  const customProperties = {}

  const generateProperties = (object, previousKey) => {
    Object.entries(object).forEach(([key, value]) => {
      let formattedKey = pluralize(key, 1)

      if (prefix && !previousKey) {
        formattedKey = `${prefix}-${formattedKey}`
      }

      const newKey = previousKey
        ? previousKey + '-' + formattedKey
        : formattedKey

      if (Array.isArray(value)) {
        value.forEach((item, i) => {
          customProperties[`--${newKey}-${i}`] = item
        })
      } else if (Object(value) === value) {
        generateProperties(value, newKey)
      } else {
        customProperties[`--${newKey}`] = value
      }
    })
  }

  generateProperties(theme)

  return customProperties
}
