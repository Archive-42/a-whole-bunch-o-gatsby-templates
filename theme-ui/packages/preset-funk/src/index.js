import base from '@theme-ui/preset-base'

export const funk = {
  ...base,
  fonts: {
    body: 'Poppins, sans-serif',
    heading: 'Poppins, sans-serif',
    monospace: 'Menlo, monospace',
  },
  lineHeights: {
    body: 1.625,
    heading: 1.25,
  },
  fontWeights: {
    body: 400,
    heading: 900,
    bold: 700,
  },
  colors: {
    ...base.colors,
    primary: '#609',
    secondary: '#306',
  },
  styles: {
    ...base.styles,
  },
}

export default funk
