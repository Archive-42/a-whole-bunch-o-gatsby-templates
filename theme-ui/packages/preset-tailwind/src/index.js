// Based on https://github.com/tailwindcss/tailwindcss/blob/master/stubs/defaultConfig.stub.js

export const breakpoints = ['40rem', '48rem', '64rem', '80rem']

export const baseColors = {
  transparent: 'transparent',
  black: '#000',
  white: '#fff',
  gray: [
    null,
    '#f7fafc',
    '#edf2f7',
    '#e2e8f0',
    '#cbd5e0',
    '#a0aec0',
    '#718096',
    '#4a5568',
    '#2d3748',
    '#1a202c',
  ],
  red: [
    null,
    '#fff5f5',
    '#fed7d7',
    '#feb2b2',
    '#fc8181',
    '#f56565',
    '#e53e3e',
    '#c53030',
    '#9b2c2c',
    '#742a2a',
  ],
  orange: [
    null,
    '#fffaf0',
    '#feebc8',
    '#fbd38d',
    '#f6ad55',
    '#ed8936',
    '#dd6b20',
    '#c05621',
    '#9c4221',
    '#7b341e',
  ],
  yellow: [
    null,
    '#fffff0',
    '#fefcbf',
    '#faf089',
    '#f6e05e',
    '#ecc94b',
    '#d69e2e',
    '#b7791f',
    '#975a16',
    '#744210',
  ],
  green: [
    null,
    '#f0fff4',
    '#c6f6d5',
    '#9ae6b4',
    '#68d391',
    '#48bb78',
    '#38a169',
    '#2f855a',
    '#276749',
    '#22543d',
  ],
  teal: [
    null,
    '#e6fffa',
    '#b2f5ea',
    '#81e6d9',
    '#4fd1c5',
    '#38b2ac',
    '#319795',
    '#2c7a7b',
    '#285e61',
    '#234e52',
  ],
  blue: [
    null,
    '#ebf8ff',
    '#bee3f8',
    '#90cdf4',
    '#63b3ed',
    '#4299e1',
    '#3182ce',
    '#2b6cb0',
    '#2c5282',
    '#2a4365',
  ],
  indigo: [
    null,
    '#ebf4ff',
    '#c3dafe',
    '#a3bffa',
    '#7f9cf5',
    '#667eea',
    '#5a67d8',
    '#4c51bf',
    '#434190',
    '#3c366b',
  ],
  purple: [
    null,
    '#faf5ff',
    '#e9d8fd',
    '#d6bcfa',
    '#b794f4',
    '#9f7aea',
    '#805ad5',
    '#6b46c1',
    '#553c9a',
    '#44337a',
  ],
  pink: [
    null,
    '#fff5f7',
    '#fed7e2',
    '#fbb6ce',
    '#f687b3',
    '#ed64a6',
    '#d53f8c',
    '#b83280',
    '#97266d',
    '#702459',
  ],
}

export const colors = {
  ...baseColors,
  grayDark: baseColors.gray[8],
  text: baseColors.gray[8],
  background: baseColors.white,
  primary: baseColors.blue[3],
  secondary: baseColors.gray[6],
  muted: baseColors.gray[3],
  success: baseColors.green[3],
  info: baseColors.blue[4],
  warning: baseColors.yellow[3],
  danger: baseColors.red[3],
  light: baseColors.gray[1],
  dark: baseColors.gray[8],
  textMuted: baseColors.gray[6],
}

export const baseFonts = {
  sans:
    '-apple-system, BlinkMacSystemFont,"Segoe UI",Roboto,"Helvetica Neue",Arial,"Noto Sans",sans-serif,"Apple Color Emoji","Segoe UI Emoji","Segoe UI Symbol","Noto Color Emoji"',
  serif: 'Georgia, Cambria, "Times New Roman", Times, serif',
  mono: 'Menlo,Monaco,Consolas,"Liberation Mono","Courier New",monospace',
}

export const fonts = {
  ...baseFonts,
  body: baseFonts.sans,
  heading: 'inherit',
  monospace: baseFonts.mono,
}

export const fontSizes = [
  '0.875rem',
  '1rem',
  '1.25rem',
  '1.5rem',
  '1.875rem',
  '2.25rem',
  '3rem',
  '4rem',
  '4.5rem',
]

export const baseFontWeights = {
  hairline: '100',
  thin: '200',
  light: '300',
  normal: '400',
  medium: '500',
  semibold: '600',
  bold: '700',
  extrabold: '800',
  black: '900',
}

export const fontWeights = {
  ...baseFontWeights,
  body: baseFontWeights.normal,
  heading: baseFontWeights.bold,
}

export const baseLineHeights = {
  none: '1',
  tight: '1.25',
  relaxed: '1.625',
  loose: '2',
}

export const lineHeights = {
  ...baseLineHeights,
  body: baseLineHeights.relaxed,
  heading: baseLineHeights.tight,
}

export const space = [
  0,
  '0.25rem',
  '0.5rem',
  '1rem',
  '2rem',
  '4rem',
  '8rem',
  '16rem',
  '32rem',
]

export const letterSpacings = {
  tight: '-0.025em',
  normal: '0',
  wide: '0.025em',
  wider: '0.05em',
  widest: '0.1em',
}

export const sizes = {
  px: '1px',
  '0': '0',
  '1': '0.25rem',
  '2': '0.5rem',
  '3': '0.75rem',
  '4': '1rem',
  '5': '1.25rem',
  '6': '1.5rem',
  '8': '2rem',
  '10': '2.5rem',
  '12': '3rem',
  '16': '4rem',
  '20': '5rem',
  '24': '6rem',
  '32': '8rem',
  '40': '10rem',
  '48': '12rem',
  '56': '14rem',
  '64': '16rem',
  xs: '20rem',
  sm: '24rem',
  md: '28rem',
  lg: '32rem',
  xl: '36rem',
  '2xl': '42rem',
  '3xl': '48rem',
  '4xl': '56rem',
  '5xl': '64rem',
  '6xl': '72rem',
}

export const borderWidths = {
  px: '1px',
  '0': '0',
  '2': '2px',
  '4': '4px',
  '8': '8px',
}

export const radii = {
  none: '0',
  sm: '0.125rem',
  default: '0.25rem',
  lg: '0.5rem',
  full: '9999px',
}

export const shadows = {
  default: '0 1px 3px 0 rgba(0, 0, 0, 0.1), 0 1px 2px 0 rgba(0, 0, 0, 0.06)',
  md: '0 4px 6px -1px rgba(0, 0, 0, 0.1), 0 2px 4px -1px rgba(0, 0, 0, 0.06)',
  lg: '0 10px 15px -3px rgba(0, 0, 0, 0.1), 0 4px 6px -2px rgba(0, 0, 0, 0.05)',
  xl:
    '0 20px 25px -5px rgba(0, 0, 0, 0.1), 0 10px 10px -5px rgba(0, 0, 0, 0.04)',
  '2xl': '0 25px 50px -12px rgba(0, 0, 0, 0.25)',
  inner: 'inset 0 2px 4px 0 rgba(0, 0, 0, 0.06)',
  outline: '0 0 0 3px rgba(66, 153, 225, 0.5)',
  none: 'none',
}

export const zIndices = {
  auto: 'auto',
  '0': '0',
  '10': '10',
  '20': '20',
  '30': '30',
  '40': '40',
  '50': '50',
}

const heading = {
  fontFamily: 'heading',
  fontWeight: 'heading',
  lineHeight: 'heading',
  m: 0,
  mb: 1,
}

export const styles = {
  root: {
    fontFamily: 'body',
    lineHeight: 'body',
    fontWeight: 'body',
  },
  a: {
    color: 'primary',
    textDecoration: 'none',
    ':hover': {
      textDecoration: 'underline',
    },
  },
  h1: {
    ...heading,
    fontSize: 6,
    mt: 2,
  },
  h2: {
    ...heading,
    fontSize: 5,
    mt: 2,
  },
  h3: {
    ...heading,
    fontSize: 4,
    mt: 3,
  },
  h4: {
    ...heading,
    fontSize: 3,
  },
  h5: {
    ...heading,
    fontSize: 2,
  },
  h6: {
    ...heading,
    fontSize: 1,
    mb: 2,
  },
  code: {},
  pre: {},
  hr: {
    bg: 'muted',
    border: 0,
    height: '1px',
    m: 3,
  },
}

export const tailwind = {
  breakpoints,
  colors,
  fonts,
  fontSizes,
  fontWeights,
  lineHeights,
  space,
  letterSpacings,
  sizes,
  borderWidths,
  radii,
  shadows,
  zIndices,
  styles,
}

export default tailwind
