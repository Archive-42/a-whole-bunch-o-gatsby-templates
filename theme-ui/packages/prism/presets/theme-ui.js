export default {
  '.comment,.prolog,.doctype,.cdata,.punctuation,.operator,.entity,.url': {
    color: 'gray',
  },
  '.comment': {
    fontStyle: 'italic',
  },
  '.property, .tag, .boolean, .number, .constant, .symbol, .deleted, .function, .class-name, .regex, .important, .variable': {
    color: 'accent',
  },
  '.atrule, .attr-value, .keyword': {
    color: 'primary',
  },
  '.selector, .attr-name, .string, .char, .builtin, .inserted': {
    color: 'secondary',
  },
}
