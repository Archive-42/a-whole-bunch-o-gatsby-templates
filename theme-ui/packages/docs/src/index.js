/** @jsx jsx */
import { jsx } from 'theme-ui'
import Layout from './components/layout'

export const wrapPageElement = ({ element, props }) => (
  <Layout
    {...props}
    fullwidth={props.location.pathname === '/'}
    children={element}
  />
)
