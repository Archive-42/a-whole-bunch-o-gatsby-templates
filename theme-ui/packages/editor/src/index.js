import React, { useReducer, useEffect } from 'react'
import { Global } from '@emotion/core'
import { ThemeProvider, Styled, ColorMode } from 'theme-ui'
import Panel from './components/Panel'
import editorTheme from './theme'
import { runScript, mergeState } from './utils'

const Editor = ({ panelColorMode }) => {
  const [state, setState] = useReducer(mergeState, {
    theme: null,
    colorMode: null,
  })

  const getTheme = () => {
    runScript(`window.__THEME_UI__.theme`).then(theme => {
      setState({ theme })
    })
  }

  const getColorMode = () => {
    runScript(`window.__THEME_UI__.colorMode`).then(colorMode => {
      setState({ colorMode })
    })
  }

  const setTheme = nextTheme => {
    const json = JSON.stringify(nextTheme)
    runScript(`window.__THEME_UI__.setTheme(${json})`)
    setState({ theme: nextTheme })
  }

  const setColorMode = nextMode => {
    runScript(`window.__THEME_UI__.setColorMode('${nextMode}')`)
  }

  useEffect(() => {
    getTheme()
    getColorMode()
  }, [])

  return (
    <ThemeProvider theme={editorTheme}>
      <Styled.root>
        <ColorMode />
        <Global
          styles={{
            'html,body': {
              margin: 0,
            },
          }}
        />
        {state.theme && (
          <Panel
            panelColorMode={panelColorMode}
            state={state}
            setTheme={setTheme}
            setColorMode={setColorMode}
          />
        )}
      </Styled.root>
    </ThemeProvider>
  )
}

export { Editor }
