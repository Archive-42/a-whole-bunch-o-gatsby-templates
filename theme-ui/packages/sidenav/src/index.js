/** @jsx jsx */
import { jsx, css, ThemeProvider } from 'theme-ui'
import { MDXProvider } from '@mdx-js/react'
import React from 'react'
import { Global } from '@emotion/core'
import merge from 'deepmerge'

const Overlay = ({ onClick }) => (
  <React.Fragment>
    <div
      onClick={onClick}
      sx={{
        position: 'fixed',
        top: 0,
        right: 0,
        bottom: 0,
        left: 0,
      }}
    />
    <Global
      styles={css({
        body: {
          overflow: ['hidden', 'auto'],
        },
      })}
    />
  </React.Fragment>
)

export const Sidenav = React.forwardRef(
  ({ open, styles = {}, components, ...props }, ref) => {
    return (
      <ThemeProvider
        theme={{
          styles: merge(
            {
              ul: {
                listStyle: 'none',
                p: 0,
                m: 0,
                ul: {
                  a: {
                    pl: 4,
                  },
                },
              },
              a: {
                display: 'block',
                px: 2,
                py: 2,
                color: 'inherit',
                textDecoration: 'none',
                fontSize: 1,
                fontWeight: 'bold',
              },
            },
            styles
          ),
        }}>
        {open && <Overlay {...props} />}
        <MDXProvider components={components}>
          <div
            {...props}
            ref={ref}
            sx={{
              position: ['fixed', 'sticky'],
              top: 0,
              left: 0,
              bottom: [0, 'auto'],
              zIndex: 1,
              minWidth: 0,
              width: 256,
              maxHeight: '100vh',
              overflowX: 'visible',
              overflowY: 'auto',
              WebkitOverflowScrolling: 'touch',
              transition: 'transform .2s ease-out',
              transform: [open ? 'translateX(0)' : 'translate(-100%)', 'none'],
              bg: ['background', 'transparent'],
            }}
          />
        </MDXProvider>
      </ThemeProvider>
    )
  }
)

const flattenLinks = children =>
  React.Children.toArray(children).reduce((acc, child) => {
    if (child.props && child.props.mdxType === 'a') {
      return [...acc, child]
    }
    if (!child.props || !child.props.children) return acc
    return React.Children.toArray([
      ...acc,
      ...flattenLinks(child.props.children),
    ])
  }, [])

const removeSlash = str => (str.length > 1 ? str.replace(/\/$/, '') : str)

const PaginationLink = ({
  label,
  children,
  mdxType,
  originalType,
  parentName,
  ...props
}) => (
  <a
    {...props}
    sx={{
      fontSize: 1,
      color: 'inherit',
      textDecoration: 'none',
      fontWeight: 'bold',
    }}>
    <div>{label}</div>
    <div
      sx={{
        fontSize: 3,
      }}>
      {children}
    </div>
  </a>
)

export const Pagination = ({ pathname = '', children, ...props }) => {
  const links = flattenLinks(children)
  const index = links.findIndex(
    link => link.props.href === removeSlash(pathname)
  )
  const hasPagination = index > -1
  const previous = links[index - 1]
  const next = links[index + 1]

  return (
    <div
      {...props}
      sx={{
        display: 'flex',
      }}>
      {hasPagination && previous && (
        <PaginationLink {...previous.props} label="Previous:" />
      )}
      <div sx={{ mx: 'auto' }} />
      {hasPagination && next && (
        <PaginationLink {...next.props} label="Next:" />
      )}
    </div>
  )
}
