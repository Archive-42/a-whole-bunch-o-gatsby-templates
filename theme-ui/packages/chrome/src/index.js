import React from 'react'
import { render } from 'react-dom'
import { Editor } from '@theme-ui/editor'

const Chrome = () => {
  const panelColorMode =
    window.chrome.devtools.panels.themeName === 'dark' ? 'dark' : 'light'

  return <Editor panelColorMode={panelColorMode} />
}

render(<Chrome />, document.getElementById('root'))
