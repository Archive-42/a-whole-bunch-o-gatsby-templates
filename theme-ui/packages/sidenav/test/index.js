/** @jsx jsx */
import { jsx } from 'theme-ui'
import renderer from 'react-test-renderer'
import { Sidenav, Pagination } from '../src'

const links = (
  <ul>
    <li>
      <a mdxType="a" href="/">
        Beep
      </a>
    </li>
    <li>
      <a mdxType="a" href="/boop">
        Boop
      </a>
    </li>
    <li>
      <a mdxType="a" href="/bop">
        Bop
      </a>
    </li>
  </ul>
)

test('Sidenav renders', () => {
  const json = renderer.create(<Sidenav>{links}</Sidenav>).toJSON()
  expect(json).toMatchSnapshot()
})

test('Sidenav renders open', () => {
  const json = renderer.create(<Sidenav open={true}>{links}</Sidenav>).toJSON()
  expect(json).toMatchSnapshot()
})

test('Pagination renders', () => {
  const json = renderer
    .create(<Pagination pathname="/" children={links} />)
    .toJSON()
})
