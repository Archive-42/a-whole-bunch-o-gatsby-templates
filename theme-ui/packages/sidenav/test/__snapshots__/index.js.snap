// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sidenav renders 1`] = `
.emotion-0 {
  position: fixed;
  top: 0;
  left: 0;
  bottom: 0;
  z-index: 1;
  min-width: 0;
  width: 256px;
  max-height: 100vh;
  overflow-x: visible;
  overflow-y: auto;
  -webkit-overflow-scrolling: touch;
  -webkit-transition: -webkit-transform .2s ease-out;
  -webkit-transition: transform .2s ease-out;
  transition: transform .2s ease-out;
  -webkit-transform: translate(-100%);
  -ms-transform: translate(-100%);
  transform: translate(-100%);
  background-color: background;
}

@media screen and (min-width:40em) {
  .emotion-0 {
    position: -webkit-sticky;
    position: sticky;
    bottom: auto;
    -webkit-transform: none;
    -ms-transform: none;
    transform: none;
    background-color: transparent;
  }
}

<div
  className="emotion-0"
>
  <ul>
    <li>
      <a
        href="/"
        mdxType="a"
      >
        Beep
      </a>
    </li>
    <li>
      <a
        href="/boop"
        mdxType="a"
      >
        Boop
      </a>
    </li>
    <li>
      <a
        href="/bop"
        mdxType="a"
      >
        Bop
      </a>
    </li>
  </ul>
</div>
`;

exports[`Sidenav renders open 1`] = `
Array [
  .emotion-0 {
  position: fixed;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
}

<div
    className="emotion-0"
  />,
  .emotion-0 {
  position: fixed;
  top: 0;
  left: 0;
  bottom: 0;
  z-index: 1;
  min-width: 0;
  width: 256px;
  max-height: 100vh;
  overflow-x: visible;
  overflow-y: auto;
  -webkit-overflow-scrolling: touch;
  -webkit-transition: -webkit-transform .2s ease-out;
  -webkit-transition: transform .2s ease-out;
  transition: transform .2s ease-out;
  -webkit-transform: translateX(0);
  -ms-transform: translateX(0);
  transform: translateX(0);
  background-color: background;
}

@media screen and (min-width:40em) {
  .emotion-0 {
    position: -webkit-sticky;
    position: sticky;
    bottom: auto;
    -webkit-transform: none;
    -ms-transform: none;
    transform: none;
    background-color: transparent;
  }
}

<div
    className="emotion-0"
  >
    <ul>
      <li>
        <a
          href="/"
          mdxType="a"
        >
          Beep
        </a>
      </li>
      <li>
        <a
          href="/boop"
          mdxType="a"
        >
          Boop
        </a>
      </li>
      <li>
        <a
          href="/bop"
          mdxType="a"
        >
          Bop
        </a>
      </li>
    </ul>
  </div>,
]
`;
